File: articles/grit-and-purpose/go/grit_purpose_score.go
package main

import "fmt"

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func PurposeScore(personalMeaning, longTermDirection, beyondSelfContribution float64) float64 {
	return 0.34*personalMeaning +
		0.33*longTermDirection +
		0.33*beyondSelfContribution
}

func PersistenceReadinessIndex(grit, purpose, socialSupport, autonomySupport, burnout float64) float64 {
	return 0.22*grit +
		0.28*purpose +
		0.18*socialSupport +
		0.18*autonomySupport -
		0.18*burnout
}

func main() {
	grit := GritScore(0.82, 0.64)
	purpose := PurposeScore(0.74, 0.82, 0.88)
	readiness := PersistenceReadinessIndex(grit, purpose, 0.58, 0.72, -0.30)

	fmt.Printf("Synthetic grit score: %.3f\n", grit)
	fmt.Printf("Synthetic purpose score: %.3f\n", purpose)
	fmt.Printf("Synthetic persistence readiness index: %.3f\n", readiness)
}

File: articles/grit-and-purpose/fortran/grit_purpose_score.f90
program grit_purpose_score
  implicit none
  real :: perseverance_effort, consistency_interests, grit
  real :: personal_meaning, long_term_direction, beyond_self_contribution, purpose
  real :: social_support, autonomy_support, burnout, readiness

  perseverance_effort = 0.82
  consistency_interests = 0.64
  personal_meaning = 0.74
  long_term_direction = 0.82
  beyond_self_contribution = 0.88
  social_support = 0.58
  autonomy_support = 0.72
  burnout = -0.30

  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests
  purpose = 0.34 * personal_meaning + 0.33 * long_term_direction + 0.33 * beyond_self_contribution
  readiness = 0.22 * grit + 0.28 * purpose + 0.18 * social_support + 0.18 * autonomy_support - 0.18 * burnout

  print *, "Synthetic grit score:", grit
  print *, "Synthetic purpose score:", purpose
  print *, "Synthetic persistence readiness index:", readiness
end program grit_purpose_score
